pulser/parametrized/paramobj.py

```diff
@@ -69,7 +71,7 @@ def __init__(self, cls, *args, **kwargs):
             kwargs: The kwargs for calling `cls`.
         """
         self.cls = cls
-        self._variables = {}
+        self._variables: Dict[str, ParamObj] = {}
```

Comment:
This is not quite right, it's rather
```suggestion
        self._variables: Dict[str, Variable] = {}
```